test/connection.spec.js

```diff
@@ -303,6 +303,26 @@ describe('connection tests', function () {
                     });
                 });
 
+                it('retry_strategy used to reconnect with object', function (done) {
+                    client = redis.createClient({
+                        retry_strategy: function (options) {
+                            if (options.total_retry_time > 150) {
+                                client.set('foo', 'bar', function (err, res) {
+                                    assert.strictEqual(err.message, 'Stream connection ended and command aborted.');
+                                    assert.strictEqual(err.code, 'NR_CLOSED');
+                                    assert.strictEqual(err.origin.code, 'ECONNREFUSED');
+                                    done();
+                                });
+                            }
+                            return {
+                                retry_delay: Math.min(options.attempt * 25, 200),
+                                error: options.error
```

Comment:
I'm not sure I follow. Do you mean just in the context of this test case, or in the implementation? With this change we have to return the error and the retry_delay, otherwise the reconnect is never attempted.